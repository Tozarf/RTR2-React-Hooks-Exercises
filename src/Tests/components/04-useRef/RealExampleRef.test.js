import React from 'react'
import { shallow } from "enzyme"
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"



describe('testing <RealExampleref/> ', () => {
    
    const wrapper = shallow(<RealExampleRef/>)
    
    test('should  match snapshot ', () => {
        
        expect(wrapper).toMatchSnapshot()
        expect( wrapper.find("MultipleCustomHooks").exists() ).toBe(false)

    })
    
    test('should show the component <MultipleCustomHooks/>', () => {
        
        wrapper.find("button").simulate("click")

        expect( wrapper.find("MultipleCustomHooks").exists() ).toBe(true)

    })
    
})
