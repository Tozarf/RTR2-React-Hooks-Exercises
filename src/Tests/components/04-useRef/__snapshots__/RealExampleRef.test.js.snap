// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <RealExampleref/>  should  match snapshot  1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
